unit/solvers/smt2_incremental/smt_terms.cpp

```diff
@@ -0,0 +1,221 @@
+// Author: Diffblue Ltd.
+
+#include <testing-utils/use_catch.h>
+
+#include <solvers/smt2_incremental/smt_terms.h>
+
+#include <util/mp_arith.h>
+
+#include <type_traits>
+
+TEST_CASE("smt_bool_literal_termt upcasts.", "[core][smt2_incremental]")
+{
+  smt_termt term{smt_bool_literal_termt{true}};
+  term = smt_bool_literal_termt{false};
+}
+
+TEST_CASE("Test smt_termt.pretty is accessible.", "[core][smt2_incremental]")
```

Comment:
Technically this is testing both that the `pretty` function is accessible and the output. I guess a change to the base `pretty` function could invalidate this test without any changes to the `smt_termt` code.